practice-app/client/pages/user/main.js

```diff
@@ -11,7 +11,11 @@ export default function main() {
     const handleClick = async (values) => {
         router.push(`/chess`);
     };
-
+    if (typeof window !== 'undefined') {
+        if(!localStorage.getItem("access_token")){
+            router.push(`/user/login`);
+        };
+    }
```

Comment:
I guess you want to redirect user if there is not access token. The correct place to do this is useEffect. You will not `if (typeof window !== 'undefined') {` there.